File: core/src/main/java/de/s42/core/Platform.java
package de.s42.core;

import org.bukkit.entity.Entity;
import org.bukkit.entity.Player;
import org.bukkit.entity.Villager;
import org.bukkit.inventory.InventoryView;
import org.bukkit.plugin.Plugin;

import java.util.function.BiFunction;

public interface Platform {

    MessageService getMessageService();

    BiFunction<Villager, Player, InventoryView> getOpenInventoryViewFunction();

    void runAsync(Plugin plugin, Runnable task);

    void runOnEntity(Plugin plugin, Entity entity, Runnable task);
}

File: core/src/main/kotlin/tr/s42/core/Platform.kt
package tr.s42.core

import org.bukkit.entity.Player
import org.bukkit.entity.Villager
import org.bukkit.inventory.InventoryView
import java.util.function.BiFunction

interface Platform {

    val messageService: MessageService
    val openInventoryViewFunction: BiFunction<Villager, Player, InventoryView>

}